### PR TITLE
wait for the OpenEBS admission server service to have endpoints

### PR DESCRIPTION
**Pull request by laverya**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: addons/openebs/template/base/install.sh [hunk 1]
```diff
@@ -157,3 +170,10 @@ function openebs_migrate_post_helm_resources() {
     # name changed from openebs-maya-operator > openebs
     kubectl delete clusterrolebinding openebs-maya-operator 2>/dev/null || true
 }
+
+function kubernetes_service_healthy() {
+    local namespace=$1
+    local name=$2
+
+    kubectl -n "$namespace" get endpoints "$name" | grep -v "NAME" | grep -v "<none>" &>/dev/null
+}
```
Comment (laverya) [inline addons/openebs/template/base/install.sh:179]: I plan to move this to the common set of functions

**Review by emosbaugh — DISMISSED** [review]

File: addons/openebs/2.12.9/install.sh [hunk 2]
```diff
@@ -95,6 +98,19 @@ function openebs_apply_storageclasses() {
     kubectl apply -k "$dst/"
 }
 
+function openebs_await_admissionserver() {
+    echo "always called for debugging purposes ****************************************************************"
```
Comment (emosbaugh) [inline addons/openebs/2.12.9/install.sh:102]: remove this

File: addons/openebs/2.12.9/install.sh [hunk 3]
```diff
@@ -95,6 +98,19 @@ function openebs_apply_storageclasses() {
...
+    kubectl get validatingWebhookConfiguration
+    if kubectl get validatingWebhookConfiguration openebs-validation-webhook-cfg &>/dev/null; then
+        logStep "Waiting for OpenEBS admission controller service to be ready"
+        spinner_until 120 kubernetes_service_healthy "$OPENEBS_NAMESPACE" admission-server-svc
+        logSuccess "OpenEBS admission controller service is ready"
+    else
+        echo "for debugging purposes ****************************************************************"
```
Comment (emosbaugh) [inline addons/openebs/2.12.9/install.sh:109]: remove this

File: addons/openebs/template/testgrid/k8s-docker-localpv.yaml [hunk 4]
```diff
@@ -48,6 +53,15 @@
...
       version: "__testver__"
       s3Override: "__testdist__"
+  postInstallScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    minio_object_store_info
+    validate_read_write_object_store rwtest testfile.txt
+  postUpgradeScript: |
+    source /opt/kurl-testgrid/testhelpers.sh
+    minio_object_store_info
+    validate_testfile rwtest testfile.txt
```
Comment (emosbaugh) [inline addons/openebs/template/testgrid/k8s-docker-localpv.yaml:63]: does this validate write after upgrade? any reason not to add validate_read_write_object_store after this?

File: addons/openebs/template/base/install.sh (hunk 1 above)
Comment (emosbaugh) [reply addons/openebs/template/base/install.sh:179]: move this to scripts/common/kuberentes.sh

Comment (emosbaugh) [reply addons/openebs/template/base/install.sh:179]: you can omit `grep -v "NAME"` by passing `--no-headers` flag to kubectl

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/openebs/2.12.9/install.sh (hunk 2 above)
Comment (laverya) [reply addons/openebs/2.12.9/install.sh:102]: oh shoot

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**